### DISCUSSION TITLE
배포 관련해 있던 이슈 논의

### DISCUSSION THREAD

**Discussion by howons** [prgrms-web-devcourse/Team-6Heroes-OneDayHero-FE#121] [General]
이야기해볼 이슈가 두 개 있습니다.

## Mock data 배포 환경에서 미적용
이건 원인을 알았는데 route handler가 사실 client component 에서만 작동하는 기능이라고 합니다. 처음 build를 할 때 server component와 동시에 route handler도 생성하는 방식이어서 server component에선 route handler의 url로 접근하기 어려운 것 같아요. dev 환경에선 요청이 들어올 때 route나 page를 컴파일하는 방식으로 동작하는 것 같은데 그래서 작동했던 것 같습니다.

이제와서 Mock 방식을 바꾸긴 번거롭기도 하고 어쩌피 mock은 개발 중에 필요한 기능이니 지금 상태로 진행해도 괜찮지 않을까 싶습니다. 혹시 다른 의견이 있다면 말씀해주세요.

## build 중 fetch 오류
이건 api가 안 열려있으니 오류나는 게 맞긴 한데 에러 바운더리 기반인 `error.tsx`를 설정해도 오류를 잡아오질 못하더라고요. 공식 문서에서는 서버에서의 오류도 잡는다고 나와있기는 한데 실제로는 클라 쪽 오류만 잡는 것 같습니다. api의 오류 하나로도 빌드를 못한다는 게 너무 불안정한 것 같아서 계속 고민해보다가 services의 fetching 함수들을 에러 처리 기능도 하도록 리펙토링 해보려합니다.

사용할 때 이런 느낌이 될 것 같아요
```tsx
const { data, isError } = getMission(missionId);

if (isError) return <ErrorPage />;

return {data.map ~~}
```

**Comment by Jeong-Taeho** [comment]
좋은 것 같습니다!! 저희 dev 환경에서는 잘되서 client component에서만 작동되는지도 몰랐네요.. 이젠 슬슬 백엔드 api를 사용해볼 때가 되서 호원님 말씀대로 이 상태로 진행해도 괜찮을 것 같습니다!! error 관련해서도 패치함수를 만들게 되면 호원님 방식을 참고해서 만들어야 겠군요.. 공식문서에서도 제대로 안나와있어서 힘드셨을텐데 고생 많으셨습니다.. 감사합니다!!!
